### ISSUE TITLE
No-Semicolon support removed in v3.1

### ISSUE THREAD

**Issue by Blasz** [styled-components/styled-components#1531]
## Version
styled-components: 3.1.6

## Reproduction

v3.0.2 (working) - https://codesandbox.io/s/6vyym18yqw
v3.1.6 (not working) - https://codesandbox.io/s/xrnom63qnz

## Steps to reproduce

Leave a trailing semicolon off of a css declaration that is not the last one in a rule.

## Expected Behavior

Semicolon is automatically added to the declaration so that it applies correctly in the browser.

## Actual Behavior

No semicolon is added resulting in the declaration and remaining declarations not being applied.

We're currently in the process of upgrading styled components in Atlaskit from v1 to v3 but the lack of no-semicolon support will introduce regressions that may fly under the radar.

**Comment by kitten** [comment]
@Blasz I'm actually seeing the same malformed CSS in both reproduction. Is there possibly something wrong with those CodeSandboxes?

```
.gtdQEe {
    width: 200px;
    height: 200px;
    border: 1px solid black;
    margin: 20px background-color:red; /* this is the rule in question */
}

.sc-bdVaJa {
}
```

So since you said it's working in 3.0.2, am I looking at the wrong thing here?

**Comment by Blasz** [comment]
Sorry guys, looks like codesandbox was caching the styled-components dependency for me after I upgraded from v1.4.6, giving me a false positive that it was working in v3.0.2.

I can confirm that the malformed css occurs in both v2.4.0 and v3.0.2 so I don't think it was ever supported with the new css processing lib.

I had a dig around before I made the ticket and came across https://github.com/thysultan/stylis.js/issues/12 and found a reference to the semicolon setting in the code (https://github.com/styled-components/styled-components/blob/v3.1.6/src/utils/stringifyRules.js#L12) but I misread it to think that it meant that no-semicolon support was enabled rather than disabled.

Is there any reason why it is disabled rather than enabled?

**Comment by kitten** [comment]
@Blasz Hm, that's a good catch. The docs we're unclear before whether `semicolon: true` meant `no-semicolon` support is enabled or disabled.

```
	// (dis/en)able (no)semicolon support
	// false to enable no-semicolons (default)
	semicolon: {Boolean},
```

This here in the stylis docs makes me think though that `true` means "disabled"? @thysultan can you confirm that we're accidentally disabling semicolon-autocompletion?

**Comment by thysultan** [comment]
Yes, it means "disabled". The code in the linked codesandbox works for me with no-semicolon option though.

**Comment by kitten** [comment]
@thisguychris @Blasz alright, then the change should be simple. I'll open a quick PR

**Comment by kitten** [comment]
@thysultan @Blasz I've got a PR for this ready: #1532. Care to take a look?

@thysultan I just remembered us having this convo before, I think, but I'd like to wholeheartedly and carefully suggest this option to be renamed, maybe to `insertMissingSemis`, or `autocompleteSemis`, which would be a lot easier to understand 😅 I think, I must've introduced this regression myself at some point, but it'd be useful if this would be less likely to occur to others.

**Comment by Blasz** [comment]
Thanks for the quick response @kitten :)
